src/structure/starboard/star.ts

```diff
@@ -0,0 +1,103 @@
+import { TextChannel } from 'discord.js';
+import { MessageEmbed } from 'discord.js';
+import { Message } from 'discord.js';
+import path from 'path';
+import { MinehutClient } from '../../client/minehutClient';
+import { StarModel } from '../../model/star';
+import { IMAGE_LINK_REGEX } from '../../util/constants';
+
+export interface StarboardData {
+	msg: Message;
+	channel: TextChannel;
+	count: number;
+	starredBy: string[];
+}
+
+export class Starboard {
+	channel: TextChannel;
+	msg: Message;
+	count: number;
+	starredBy: string[];
+
+	constructor(data: StarboardData) {
+		this.channel = data.channel;
+		this.msg = data.msg;
+		this.count = data.count;
+		this.starredBy = data.starredBy;
+	}
+
+	async add() {
+		const member = this.msg.member;
+		const embed = new MessageEmbed()
+			.setColor('YELLOW')
+			.setTimestamp()
+			.setAuthor(member?.displayName, this.msg.author.displayAvatarURL());
+		if (this.msg.content) {
+			embed.setDescription(`${this.msg.content}\n\n[Jump!](${this.msg.url})`);
+		} else {
+			embed.setDescription(`[Jump!](${this.msg.url})`);
+		}
+		const img = Starboard.findImg(this.msg);
+		if (img) embed.setImage(img);
+
+		let sent = await this.channel.send(
+			`⭐**${this.count}** ${this.msg.channel} `,
+			embed
+		);
+		await StarModel.create({
+			_id: this.msg.id,
+			author: this.msg.author.id,
+			guild: this.msg.guild!.id,
+			storageMsg: sent.id,
+			starAmount: this.count,
+			starredBy: this.starredBy,
+		});
+	}
+
+	static async exists(id: string) {
+		return await StarModel.exists({ _id: id });
+	}
+
+	static emojiEquals(x: any, y: any) {
```

Comment:
What is the purpose of this function?

Reply:
If you're tlaking about Starboard#exists, I don't exactly remember why I have it so I removed it.